satellite/overlay/keylock_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package overlay_test
+
+import (
+	"strconv"
+	"testing"
+
+	"storj.io/storj/internal/teststorj"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/overlay"
+)
+
+func TestKeyLock(t *testing.T) {
+	ml := overlay.NewKeyLock()
+	key := teststorj.NodeIDFromString("hi")
+	unlockFunc := ml.Lock(key)
+	unlockFunc()
+	unlockFunc = ml.RLock(key)
+	unlockFunc()
+}
+
+func BenchmarkKeyLock(b *testing.B) {
+	b.ReportAllocs()
+	ml := overlay.NewKeyLock()
+	numNodes := 100000
+	nodeIDs := make([]storj.NodeID, numNodes)
+	for i := 0; i < numNodes; i++ {
+		nodeIDs[i] = teststorj.NodeIDFromString(strconv.Itoa(i))
```

Comment:
Use `testrand.NodeID()` rather than a sequence of items.

Reply:
I was intentionally trying to simulate the case where it has to lazily create the locks, but then is able to re-use existing locks. That is why I didn't create a random nodeID for each test iteration. Is there a better way to do this? Or should I not worry about simulating that?

Reply:
I just mean having the node-ids themselves be random rather than `teststorj.NodeIDFromString`. We have an `internal/testrand` package that can generate random node id. The slice of id-s is fine.